package l2p.gameserver.serverpackets;

import l2p.gameserver.model.L2Player;
import l2p.gameserver.model.instances.L2HennaInstance;

public class HennaInfo extends L2GameServerPacket
{
	private final L2HennaInstance[] _hennas = new L2HennaInstance[3];
	private int _count, _str, _con, _dex, _int, _wit, _men;

	public HennaInfo(L2Player player)
	{
		int j = 0;
		for(int i = 0; i < 3; i++)
		{
			L2HennaInstance h = player.getHenna(i + 1);
			if(h != null)
				_hennas[j++] = h;
		}
		_count = j;

		_str = player.getHennaStatSTR();
		_con = player.getHennaStatCON();
		_dex = player.getHennaStatDEX();
		_int = player.getHennaStatINT();
		_wit = player.getHennaStatWIT();
		_men = player.getHennaStatMEN();
	}

	@Override
	protected final void writeImpl()
	{
		writeH(_int); //equip INT
		writeH(_str); //equip STR
		writeH(_con); //equip CON
		writeH(_men); //equip MEM
		writeH(_dex); //equip DEX
		writeH(_wit); //equip WIT
		writeH(0x00); //equip LUC
		writeH(0x00); //equip CHA
		writeD(3); //interlude, slots?
		writeD(_count);
		for(int i = 0; i < _count; i++)
			if(_hennas[i] != null)
			{
				writeD(_hennas[i].getSymbolId());
				writeD(_hennas[i].getSymbolId());
			}

		writeD(0x00);	// Premium symbol ID
		writeD(0x00);	// Premium symbol left time
		writeD(0x00);	// Premium symbol active
	}

	@Override
	protected final void writeImplIT()
	{
		writeC(_int); //equip INT
		writeC(_str); //equip STR
		writeC(_con); //equip CON
		writeC(_men); //equip MEM
		writeC(_dex); //equip DEX
		writeC(_wit); //equip WIT
		writeD(3); //interlude, slots?
		writeD(_count);
		for(int i = 0; i < _count; i++)
			if(_hennas[i] != null)
			{
				writeD(_hennas[i].getSymbolId());
				writeD(_hennas[i].getSymbolId());
			}
	}
}